code-challenges.test.js

```diff
@@ -16,31 +16,134 @@
+  });
+  return sentence;
+};
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+// Snapshots:   0 total
+// Time:        0.157 s
+// Ran all test suites.
+// ✨  Done in 0.55s.
+
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false];
 // Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true];
 // Expected output: [ 2, 1, -1 ]
 
+describe("remainingNumbers", () => {
+  it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.", () => {
+    expect(remainingNumbers(hodgepodge1)).toEqual([2, 0, -1, 0]);
+    expect(remainingNumbers(hodgepodge2)).toEqual([2, 1, -1]);
+  });
+});
+
+// ReferenceError: remainingNumbers is not defined, good fail, lets make a function to make this test passsssssss....
+
 // b) Create the function that makes the test pass.
 
-// --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
+// b) Create the function that makes the test pass.
+// we need a function that takes in an array and checks for division by three
+// first thing we need to do is filter out all of the non-number elements. We can use filter method
+// once we have the arrays with just numbers, we can chain .map method and divide each number using mod operator
+// resulting array will have division by 3 remainders as elements
 
-// a) Create a test with an expect statement using the variables provided.
+// First thing is first.. we have to figure out how to select only the numbers of the mixed data array.
+// We'll use filter() on the elements and select using type() we will select the elements that are strictly
+// equal to 'number' followed by a map(elements) to return the new array of numbers divided by 3.
+
+const remainingNumbers = (array) =>
+  array.filter((elements) => 
+  typeof elements === 'number')
+  .map((elements) => elements % 3);
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.19 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 0.97s.
+
+
+// // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
+
+// // a) Create a test with an expect statement using the variables provided.
+
+
+// The instructions ar e confusing... the output is 99.. but the instructions asked for the SUM of ALL the numbers... cubed. This means that you have to add the numbers in teh array first and then cube it... you know what i mean? so the output shouldn't 99 or 
```

Comment:
Great job checking the expected output to clarify the intent of the question.  Looks like you got it!